src/mysql_vm_helpers.py

```diff
@@ -217,8 +218,9 @@ def uninstall_mysql() -> None:
                 # uninstalls fail due to SNAP_DATA_DIR fails to umount
                 # try umount it, without check
                 subprocess.run(["umount", CHARMED_MYSQL_COMMON_DIRECTORY])
+                shutil.rmtree(f"{CHARMED_MYSQL_DATA_DIRECTORY}/etc", ignore_errors=True)
                 logger.exception(f"Failed to uninstall MySQL on {attempt=}")
-        raise MySQLUninstallError
+        raise MySQLUninstallError from None
```

Comment:
Solely because the code for handling an incompatible data_dir case is done within an exception handling, so the exception messages where stacked making it harder to parse.